plugins/module_utils/config_utils.py

```diff
@@ -99,3 +104,192 @@ def changed(self) -> bool:
         """
         orig_dict = self._parse_config_from_file()
         return orig_dict != self._config_dict
+
+    def _search_map(self, dictionary, key):
+        """
+        Recursively search for a key in a nested dictionary.
+
+        This function traverses through a nested dictionary structure to find the value associated with the
+        specified key. It searches through all levels of nested dictionaries until it finds the key or
+        exhausts all possibilities.
+
+        Parameters:
+        - dictionary (dict): The dictionary to search through. This can be a multi-level nested dictionary.
+        - key (str): The key to search for in the dictionary.
+
+        Returns:
+        - The value associated with the found key, or None if the key is not found in the dictionary.
+
+        Example:
+        - Given the dictionary `{"interfaces": {"wan": {"if": 'interfaces/wan/if'}}}` and the key `'if'`,
+        the function will return `'interfaces/wan/if'`.
+        """
+
+        if isinstance(dictionary, dict):
+            for k, v in dictionary.items():
+                if k == key:
+                    return v
+                else:
+                    result = self._search_map(v, key)
+                    if result is not None:
+                        return result
+
+    def _get_xpath(self, module: str = None, setting: str = None, map_dict=None):
```

Comment:
Same thing here regarding typing and returns. Regarding the params: are they optional intentionally?

Reply:
I absolutely agree with you and thank you for pointing out this inconsistency. I have ensured consistent typing and optional parameters across the functions.